From 2988420fc041d0cb8e43ed7490208378ecbb1ec9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:05 -0400
Subject: [tools] perf/kvm: Move code to generate filename for perf-kvm to function

Message-id: <1408953210-26343-118-git-send-email-jolsa@redhat.com>
Patchwork-id: 88821
O-Subject: [PATCH RHEL7.1 BZ1133083 117/322] perf kvm: Move code to generate filename for perf-kvm to function.
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit e1a2b174dbbe08dce12bde9f05f64dbbae652bed
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Fri Dec 6 17:25:51 2013 -0500

description
===========
The code in builtin-kvm.c to generate filename for perf-kvm is useful to
other command such as builtin-diff.

This patch move the related code form builtin-kvm.c to util/util.c and
wrap them in a function named get_filename_for_perf_kvm.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-kvm.c b/tools/perf/builtin-kvm.c
index c2e5d56..c6fa3cb 100644
--- a/tools/perf/builtin-kvm.c
+++ b/tools/perf/builtin-kvm.c
@@ -1713,12 +1713,7 @@ int cmd_kvm(int argc, const char **argv, const char *prefix __maybe_unused)
   perf_guest = 1;
 
  if (!file_name) {
-  if (perf_host && !perf_guest)
-   file_name = strdup("perf.data.host");
-  else if (!perf_host && perf_guest)
-   file_name = strdup("perf.data.guest");
-  else
-   file_name = strdup("perf.data.kvm");
+  file_name = get_filename_for_perf_kvm();
 
   if (!file_name) {
    pr_err("Failed to allocate memory for filename\n");
diff --git a/tools/perf/util/util.c b/tools/perf/util/util.c
index bae8756..4a57609 100644
--- a/tools/perf/util/util.c
+++ b/tools/perf/util/util.c
@@ -482,3 +482,17 @@ int filename__read_str(const char *filename, char **buf, size_t *sizep)
  close(fd);
  return err;
 }
+
+const char *get_filename_for_perf_kvm(void)
+{
+ const char *filename;
+
+ if (perf_host && !perf_guest)
+  filename = strdup("perf.data.host");
+ else if (!perf_host && perf_guest)
+  filename = strdup("perf.data.guest");
+ else
+  filename = strdup("perf.data.kvm");
+
+ return filename;
+}
diff --git a/tools/perf/util/util.h b/tools/perf/util/util.h
index 659abf3..0171213 100644
--- a/tools/perf/util/util.h
+++ b/tools/perf/util/util.h
@@ -321,4 +321,6 @@ void free_srcline(char *srcline);
 
 int filename__read_int(const char *filename, int *value);
 int filename__read_str(const char *filename, char **buf, size_t *sizep);
+
+const char *get_filename_for_perf_kvm(void);
 #endif /* GIT_COMPAT_UTIL_H */
-- 
1.7.1